src/state/pwallet/concentratedLiquidity/hooks/evm.ts

```diff
@@ -94,17 +95,46 @@ export function useGetUserPositions() {
     if (account) {
       return tokenIdResults
         .map(({ result }) => result)
-        .filter((result): result is any => !!result) // any => CallStateResult
+        .filter((result): result is any => !!result)
         .map((result) => BigNumber.from(result[0]));
     }
     return [];
   }, [account, tokenIdResults]);
 
   const { positions, loading: positionsLoading } = useConcentratedPositionsFromTokenIds(tokenIds);
 
+  const uniqueTokens = useMemo(() => {
+    if (positions) {
+      const tokens = positions.map((position) => {
+        return [position.token0, position.token1];
+      });
+
+      const uniqueTokens = [...new Set(tokens.flat())];
+      return uniqueTokens;
+    }
+    return [];
+  }, [positions]);
+
+  const uniqueTokensWithData = useTokens(uniqueTokens);
```

Comment:
there is some type error here, please run `./node_modules/.bin/tsc` to get more details.

Reply:
done